lib/openstudio-standards/standards/Standards.Model.rb

```diff
@@ -6844,6 +6958,60 @@ def air_loop_hvac_supply_air_temperature_reset_type(air_loop_hvac)
     return 'warmest_zone'
   end
 
+  # Calculate the wwr of a surface
+  #
+  # @param surface [OpenStudio::Model::Surface]
+  # @return [Float] window to wall ratio of a surface
+  def get_wwr_of_a_surface(surface)
+    surface_area = surface.grossArea
+    surface_fene_area = 0.0
+    surface.subSurfaces.sort.each do |ss|
+      next unless ss.subSurfaceType == 'FixedWindow' || ss.subSurfaceType == 'OperableWindow' || ss.subSurfaceType == 'GlassDoor'
+
+      surface_fene_area += ss.netArea
+    end
+    return surface_fene_area / surface_area
+  end
+
+  # Calculate the door ratio of a surface
+  #
+  # @param surface [OpenStudio::Model::Surface]
+  # @return [Float] window to wall ratio of a surface
+  def get_door_ratio_of_a_surface(surface)
+    surface_area = surface.grossArea
+    surface_door_area = 0.0
+    surface.subSurfaces.sort.each do |ss|
+      next unless ss.subSurfaceType == 'Door'
+
+      surface_door_area += ss.netArea
+    end
+    return surface_door_area / surface_area
+  end
+
+  # Calculate the window to wall ratio reduction factor
+  #
+  # @param multiplier [Float] multiplier of the wwr
+  # @param surface_wwr [Float] the surface window to wall ratio
+  # @param surface_dr [Float] the surface door to wall ratio
+  # @param wwr_building_type[String] building type for wwr
+  # @param wwr_target [Float] target window to wall ratio
+  # @param total_wall_m2 [Float] total wall area of the category in m2.
+  # @param total_wall_with_fene_m2 [Float] total wall area of the category with fenestrations in m2.
+  # @param total_fene_m2 [Float] total fenestration area
+  # @param total_plenum_wall_m2 [Float] total sqaure meter of a plenum
+  # @return [Float] reduction factor
+  def get_wwr_reduction_ratio(multiplier,
+                              surface_wwr: 0.0,
+                              surface_dr: 0.0,
+                              wwr_building_type: 'All others',
+                              wwr_target: 0.0,
+                              total_wall_m2: 0.0,
+                              total_wall_with_fene_m2: 0.0,
+                              total_fene_m2: 0.0,
+                              total_plenum_wall_m2: 0.0)
+    return 1.0 - multiplier
+  end
```

Comment:
Perhaps we should rename the method name to be `model_get_wwr_reduction_ratio` to be consistent with other methods in this file.